pkg/quota/admission/runonceduration/api/types.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// RunOnceDurationConfig is the configuration for the RunOnceDuration plugin.
+// It specifies a default override value for ActiveDeadlineSeconds for a run-once pod.
+// The project that contains the pod may specify a different setting. That setting will
+// take precedence over the one configured for the plugin here.
+type RunOnceDurationConfig struct {
+	unversioned.TypeMeta
+
+	// ActiveDeadlineSecondsOverride is the value to set on containers of run-once pods
+	// Only a positive value is valid. Absence of a value means that the plugin
+	// won't make any changes to the pod
+	ActiveDeadlineSecondsOverride *int64
+
+	// AllowProjectAnnotationOverride indicates whether an override can be set in a project
+	// annotation. The project annotation override will take precedence over the globally configured
+	// override.
+	AllowProjectAnnotationOverride bool
```

Comment:
In my plugin I used the "Enabled" member to short-circuit the plugin; it's certainly a matter of taste, but it seems clearer to me if that's the purpose.